### PR TITLE
DEPR: enforce deprecation MultiIndex.is_lexsorted and MultiIndex.lexsort_depth

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

Deprecation introduced in #38701

### CODE REVIEW

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -177,6 +177,7 @@ Removal of prior version deprecations/changes
 - Remove arguments ``names`` and ``dtype`` from :meth:`Index.copy` and ``levels`` and ``codes`` from :meth:`MultiIndex.copy` (:issue:`35853`, :issue:`36685`)
 - Remove argument ``inplace`` from :meth:`MultiIndex.set_levels` and :meth:`MultiIndex.set_codes` (:issue:`35626`)
 - Disallow passing positional arguments to :meth:`MultiIndex.set_levels` and :meth:`MultiIndex.set_codes` (:issue:`41485`)
+- Removed deprecated :meth:`MultiIndex.is_lexsorted` and :meth:`MultiIndex.lexsort_depth` (:issue:`38701`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.0.0.rst:180]: Can you remove "deprecated"?

**Review by phofl — COMMENTED** [review]
Doc build fails

File: doc/source/whatsnew/v2.0.0.rst (hunk 1 above)
Comment (lukemanley) [reply doc/source/whatsnew/v2.0.0.rst]: removed it

**Comment by lukemanley** [comment]
> Doc build fails

fixed and now green. thx

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @lukemanley (can you update deprecations issue?)

**Pull request merged**